libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -163,6 +168,44 @@ impl<'a> Validator<'a> {
         errors.ok()
     }
 
+    fn validate_names_for_indexes(
+        &self,
+        ast_schema: &ast::SchemaAst,
+        schema: &dml::Datamodel,
+    ) -> Result<(), ErrorCollection> {
+        let mut errors = ErrorCollection::new();
+        let mut index_names = HashSet::new();
+
+        let multiple_indexes_with_same_name_are_supported = match self.source {
```

Comment:
This would be cleaner as

```rust
let long_variable = self.source.map(|source| source.combined_connector.supports_multiple_indexes_with_same_name()).unwrap_or(false);
```